sound/soc/sof/sof-audio.c

```diff
@@ -282,9 +283,11 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 
 	/* unprepare all widgets in the sink paths */
 	snd_soc_dapm_widget_for_each_sink_path(widget, p) {
+		if (!widget_in_list(list, p->sink))
+			continue;
```

Comment:
How this happens?
We walk from source to sink, so walk through the widgets which are in the path.
How we are enabling a widget (which is **in** the path) which is **not in** the path?

Reply:
@ujfalusi snd_soc_dapm_widget_for_each_sink_path() looks at all paths leaving the widget not just the ones that have an active DAPM widget associated with the PCM.

Reply:
@ranj063, oh yes, I see now, thanks!